src/deepsparse/transformers/pipelines_cli.py

```diff
@@ -0,0 +1,276 @@
+
+
+__all__ = [
+    "cli",
+]
+
+MODEL_DIR = os.getenv("MODEL_DIR") or ""
+
+
+def _parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Cli utility for one shot inference using pipelines"
+    )
+
+    parser.add_argument(
+        "-t",
+        "--task",
+        help="Name of the task to define which pipeline to create."
+        f"Currently supported tasks {SUPPORTED_TASKS.keys()}",
+        choices=SUPPORTED_TASKS.keys(),
+        type=str,
+        default="sentiment-analysis",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--data",
+        help="Path to file containing data for inferences, "
+        "inputs should be separated via newline",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--model-name",
+        "--model_name",
+        help="Canonical name of the hugging face model this model is based on",
+        default=None,
+        type=Optional[str],
+    )
+
+    default_model_path = os.path.join(MODEL_DIR, "model.onnx")
+    parser.add_argument(
+        "--model-path",
+        "--model_path",
+        help="Path to (ONNX) model file to run, can also be a SparseZoo stub."
+        f"Defaults to {default_model_path}",
+        default=default_model_path,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--engine-type",
+        "--engine_type",
+        help="Inference engine name to use. Supported options are 'deepsparse'"
+        "and 'onnxruntime'",
+        type=str,
+        choices=SUPPORTED_ENGINES,
+        default=SUPPORTED_ENGINES[0],
+    )
+
+    parser.add_argument(
+        "--config",
+        help="Huggingface model config, if none provided, default will be used"
+        "which will be from the model name or sparsezoo stub if given for "
+        "model path",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--tokenizer",
+        help="Huggingface tokenizer, if none provided, default will be used",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--max-length",
+        "--max_length",
+        help="Maximum sequence length of model inputs. default is 128",
+        type=int,
+        default=128,
+    )
+
+    parser.add_argument(
+        "--num-cores",
+        "--num_cores",
+        help="Number of CPU cores to run engine with. Default is the maximum "
+        "available",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        "--batch_size",
+        help="The batch size to use for pipelines",
+        type=int,
+        default=1,
+    )
+
+    scheduler_choices = ["multi", "single"]
+    parser.add_argument(
+        "--scheduler",
+        choices=scheduler_choices,
+        help="The scheduler to use for the engine. Can be None, single or multi",
+        default=scheduler_choices[0],
+    )
+
+    parser.add_argument(
+        "-o",
```

Comment:
as noted below, let's refactor this to a json file name (or if we want to open it this way, then write below using `json.dump`)